java/vector/src/main/java/org/apache/arrow/vector/ipc/message/MessageSerializer.java

```diff
@@ -403,16 +421,29 @@ public static ArrowRecordBatch deserializeRecordBatch(RecordBatch recordBatchFB,
       nodes.add(new ArrowFieldNode(node.length(), node.nullCount()));
     }
     List<ArrowBuf> buffers = new ArrayList<>();
+    long curOffset = 0L;
     for (int i = 0; i < recordBatchFB.buffersLength(); ++i) {
       Buffer bufferFB = recordBatchFB.buffers(i);
       ArrowBuf vectorBuffer = body.slice(bufferFB.offset(), bufferFB.length());
+      curOffset = bufferFB.offset() + bufferFB.length();
       buffers.add(vectorBuffer);
     }
+
+    if (curOffset % 8 != 0) {
+      curOffset += 8 - curOffset % 8;
+    }
+    ArrowBodyCompression bodyCompression = null;
```

Comment:
It looks like this is manually parsing the buffer stream?  I would have expected to directly reference A body compression object directly

Reply:
In the revised PR, we no longer parse the buffer stream.